source/extensions/filters/http/rbac/rbac_filter.cc

```diff
@@ -109,13 +109,14 @@ RoleBasedAccessControlFilter::decodeHeaders(Http::RequestHeaderMap& headers, boo
   const auto engine =
       config_->engine(callbacks_->route(), Filters::Common::RBAC::EnforcementMode::Enforced);
   if (engine != nullptr) {
+    std::string effective_policy_id;
     if (engine->handleAction(*callbacks_->connection(), headers, callbacks_->streamInfo(),
-                             nullptr)) {
+                             &effective_policy_id)) {
       ENVOY_LOG(debug, "enforced allowed");
       config_->stats().allowed_.inc();
       return Http::FilterHeadersStatus::Continue;
     } else {
-      ENVOY_LOG(debug, "enforced denied");
+      ENVOY_LOG(debug, "enforced denied {}", effective_policy_id);
```

Comment:
The effective_policy_id is the name of the matched policy, I think you can print it in both allow and deny cases:
```cpp
std::string effective_policy_id = "none";
ENVOY_LOG(debug, "enforced allowed/denied, matched policy {}", effective_policy_id);
```

Reply:
Done!